sbncode/SBNEventWeight/Calculators/BNBFlux/FluxCalcPrep.h

```diff
@@ -0,0 +1,91 @@
+// Based on  ubsim / EventWeight / Calculators / FluxUnisimWeightCalc.cxx  @ UBOONE_SUITE_v08_00_00_55 
+// Ported to/adapted for SBNCode by Keng Lin July 2021
+
+#include "art/Framework/Principal/Event.h"
+#include "nugen/NuReweight/art/NuReweight.h"
+#include "nusimdata/SimulationBase/MCFlux.h" 
+#include "nusimdata/SimulationBase/MCTruth.h"
+
+#include "sbncode/SBNEventWeight/Base/WeightCalc.h"
+#include "sbncode/SBNEventWeight/Base/WeightCalcCreator.h"
+#include "sbncode/SBNEventWeight/Base/SmearingUtils.h"//MultiGaussianSmearing!
+#include <sys/stat.h> //for exit(0); debugging purpose
+
+#include "TH1F.h"
+#include "TFile.h"
+#include "TDecompChol.h"//for Choleskey Decomposition
+
+namespace sbn {
+  namespace evwgh {
+
+    class FluxWeightCalc : public WeightCalc {
+      public:
+        FluxWeightCalc() : WeightCalc() {}
+
+        //Read FHiCL and store the settings for the reweighting environment and the calculator.
+        void Configure(fhicl::ParameterSet const& pset,
+            CLHEP::HepRandomEngine& engine) override;
+
+
+        //GetWeight() returns the final weights as a vector
+        //  each weight evaluaed by *WeightCalc() function;
+        //inu - the ith parameter.
+        std::vector<float> GetWeight(art::Event& e, size_t inu) override;
```

Comment:
offhand I feel like weights should be `double`s, any reason we think that `float`s are OK?

Reply:
A quick reference: https://hackr.io/blog/float-vs-double

The table shows that..


|Floating point type | Memory requirement | Range|
|-- | -- | --|
|Float | 4 bytes | ±3.40282347E+38F i.e. 6-7 significant digits|
|Double | 8 bytes | ±1.79769313486231570E+308 i.e. 15-16 significant digits|

It looks like 6 significant digits are sufficient for weights?